emhttp/plugins/dynamix/agents/tests/update_php_file_confinement_test.sh

```
@@ -0,0 +1,75 @@
+fails=0
+
+# Run the real update.php with a given '#file' value; echoes "written" if the
+# target was created (vulnerable) or "blocked" if confinement refused it.
+run_update() {
+  local file_value="$1"
+  ( cd "$emhttp_dir" && php -d short_open_tag=On -r '
+      error_reporting(0);
+      $_SERVER["DOCUMENT_ROOT"] = getcwd();
+      $_POST = ["#file" => $argv[1], "#raw_file" => "true", "marker" => "probe"];
+      include "update.php";
+    ' "$file_value" ) >/dev/null 2>&1 || true
```

Shellcheck (0.11.0)

[info] Expressions don't expand in single quotes, use double quotes for that.

(SC2016)